ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -1440,64 +1441,112 @@ template<class FluidSystem,class Scalar>
+        std::partial_sum(recvLenPd.begin(), recvLenPd.end(), displPd.begin() + 1);
+    }
+
+    std::vector<int> globalFailedCellsPb, globalFailedCellsPd;
     if (comm.rank() == 0) {
         globalFailedCellsPb.resize(displPb.back());
         globalFailedCellsPd.resize(displPd.back());
     }
 
-    comm.gatherv(failedCellsPb_.data(), static_cast<int>(failedCellsPb_.size()),
-                 globalFailedCellsPb.data(), recvLenPb.data(),
-                 displPb.data(), 0);
-    comm.gatherv(failedCellsPd_.data(), static_cast<int>(failedCellsPd_.size()),
-                 globalFailedCellsPd.data(),  recvLenPd.data(),
-                 displPd.data(), 0);
+    comm.gatherv(failedCellsPb_.data(),
+                 static_cast<int>(failedCellsPb_.size()),
+                 globalFailedCellsPb.data(),
+                 recvLenPb.data(),
+                 displPb.data(),
+                 0);
+
+    comm.gatherv(failedCellsPd_.data(),
+                 static_cast<int>(failedCellsPd_.size()),
+                 globalFailedCellsPd.data(),
+                 recvLenPd.data(),
+                 displPd.data(),
+                 0);
+
+    if (globalFailedCellsPb.empty() && globalFailedCellsPd.empty()) {
+        return;
+    }
+
     std::sort(globalFailedCellsPb.begin(), globalFailedCellsPb.end());
     std::sort(globalFailedCellsPd.begin(), globalFailedCellsPd.end());
 
+    auto ijkString = [this](const std::size_t globalIndex)
+    {
+        const auto ijk = this->eclState_.gridDims().getIJK(globalIndex);
+
+        return fmt::format("({},{},{})", ijk[0] + 1, ijk[1] + 1, ijk[2] + 1);
+    };
+
+    const auto maxNumCellsFaillog = static_cast<std::size_t>(20);
+
     if (!globalFailedCellsPb.empty()) {
+        const auto u = std::unique(globalFailedCellsPb.begin(),
+                                   globalFailedCellsPb.end());
+
+        const auto numFailed = static_cast<std::size_t>
+            (std::distance(globalFailedCellsPb.begin(), u));
+
         std::stringstream errlog;
-        errlog << "Finding the bubble point pressure failed for " << globalFailedCellsPb.size() << " cells [";
-        errlog << globalFailedCellsPb[0];
-        const size_t maxElems = std::min(maxNumCellsFaillog, globalFailedCellsPb.size());
-        for (size_t i = 1; i < maxElems; ++i) {
-            errlog << ", " << globalFailedCellsPb[i];
+        errlog << "Finding the bubble point pressure failed for "
+               << numFailed << " cell"
+               << ((numFailed != std::size_t{1}) ? "s" : "")
+               << " [" << ijkString(globalFailedCellsPb[0]);
+
+        const auto maxElems = std::min(maxNumCellsFaillog, numFailed);
+        for (auto i = 1 + 0*maxNumCellsFaillog; i < maxElems; ++i) {
+            errlog << ", " << ijkString(globalFailedCellsPb[i]);
         }
-        if (globalFailedCellsPb.size() > maxNumCellsFaillog) {
+
+        if (numFailed > maxNumCellsFaillog) {
```

Comment:
One could save one iteration over the vector and some memory copying  by first printing the cell indices (and simply skipping multiple) while keeping count of the number and then the total number.

Reply:
> One could save one iteration over the vector and some memory copying by first printing the cell indices (and simply skipping multiple) while keeping count of the number and then the total number.

It's a tradeoff.  These arrays **typically** don't have more than 300 elements including duplicates so I don't expect `std::unique()` to be a big cost, but I'll sketch out a proposal.  For what it's worth, the largest arrays I ever saw on the model I used for testing this was 350 elements with the common case being less than 100.

Reply:
I'll back off.